### PR TITLE
Add changeset and fix type exports

### PR DESCRIPTION
**Pull request by dale-french**
* Exports `ToggleIdType` and `VariantType`
* Adds [changesets](https://github.com/changesets/changesets) to automate versioning and releasing

### CODE REVIEW

**Comment by dale-french** [comment]
@jopdeklein please can you assist with the following:

- [ ] Add [changeset-bot](https://github.com/changesets/bot) to this repo
- [ ] Generate and add `NPM_TOKEN` to secrets under repo settings
- [ ] Under settings > general > actions
   - Give workflow read and write permissions
   - Allow GitHub actions to create and approve pull requests

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,64 +1,54 @@
...
 
-> **Tags:**
->
-> - :boom: [Breaking Change]
-> - :eyeglasses: [Spec Compliancy]
-> - :rocket: [New Feature]
-> - :bug: [Bug Fix]
-> - :memo: [Documentation]
-> - :house: [Internal]
-> - :nail_care: [Polish]
```
Comment (dale-french) [inline CHANGELOG.md:11]: Seems we will need to move this somewhere else - changesets is not smart enough to add changes below this. I'm sure there is a way but we can figure that out later unless you are sold on these being here 🙂

Comment (jopdeklein) [reply CHANGELOG.md:11]: Yeah nice to have so I think the benefits of auto-generating the changelogs outweighs the legend :)

File: src/integrations/optimizely.ts [hunk 2]
```diff
@@ -366,3 +366,8 @@ export const getEnabledFeatures = () => {
     audienceSegmentationAttributes
   )
 }
+
+/**
+ * Export imported types
+ */
+export {ToggleIdType, VariantType}
```
Comment (jopdeklein) [inline src/integrations/optimizely.ts:373]: For sake of consistency I think we can release this as 4.0.3.

File: src/integrations/optimizely.ts [hunk 3]
```diff
@@ -332,13 +332,13 @@ const convertBooleanToggleToFeatureVariant = (toggleId: ToggleIdType) => {
...
-export function toggle<A = unknown>(toggleId: string, ...variants: A[]): A
+export function toggle<A = unknown>(toggleId: ToggleIdType, ...variants: A[]): A
 export function toggle(
-  toggleId: string,
+  toggleId: ToggleIdType,
   ...variants: (() => unknown)[]
 ): unknown
 export function toggle<A = unknown>(
-  experimentId: string,
+  toggleId: ToggleIdType,
```
Comment (jopdeklein) [inline src/integrations/optimizely.ts:341]: Nice catch of inconsistent naming :)

**Review by jopdeklein — APPROVED** [review]

File: src/integrations/optimizely.ts (hunk 2 above)
Comment (dale-french) [reply src/integrations/optimizely.ts:373]: Will do. Just want to have the bot set up first before adding a changeset just to ensure that it works correctly :)

**Pull request closed without merging**